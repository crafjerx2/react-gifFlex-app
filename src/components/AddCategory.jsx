import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onAddCategory }) => {
  const [inputValue, setinputValue] = useState('');

  const onInputChange = ({ target }) => {
    setinputValue(target.value);
  };

  const onNewInput = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;

    onAddCategory(inputValue.trim());
    setinputValue('');
  };

  return (
    <form onSubmit={onNewInput} aria-label='form'>
      <input
        type='text'
        placeholder='Buscar Gifs'
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired
};
