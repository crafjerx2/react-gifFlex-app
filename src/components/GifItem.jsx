import React from 'react';
import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {
  return (
    <div className='card'>
      <img data-testid="test-img" src={url} alt={title} />
      <p data-testid="test-title">{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};
