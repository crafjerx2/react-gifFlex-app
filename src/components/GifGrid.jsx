import { GifItem } from './GifItem';
import { useFetchGifs } from '../useFetchGifs';

export const GifGrid = ({ category, onDelete }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <div className='header'>
        <h1>{category} </h1>
        <div>
          <button className='btn-delete' onClick={() => { onDelete(category); }}>Delete</button>
        </div>
      </div>
      {
        isLoading && (<h2 className='loading'>Loading....</h2>)
      }
      <div className='card-grid'>
        {
          images.map((imagen) => (
            <GifItem
              key={imagen.id}
              {...imagen}
            />
          ))
        }
      </div>
    </>
  );
};
