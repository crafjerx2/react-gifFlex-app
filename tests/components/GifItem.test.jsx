import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('GifItem test suite', () => {
  const title = 'One piece gif';
  const url = 'https://media4.giphy.com/media/4OV1bLOIWwIXRxpXlN/giphy.gif?cid=acacc881vaklavil45f31ohfkaui8zwofj1lxi7nsfnuv0y3&ep=v1_gifs_search&rid=giphy.gif&ct=g';

  test('should match the snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test('The image should has the correct url and alt', () => {
    render(<GifItem title={title} url={url} />);

    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('The p should has the correct title', () => {
    render(<GifItem title={title} url={url} />);

    //expect(screen.getByText(title)).toBeTruthy();
    expect(screen.getByTestId('test-title').innerHTML).toBe(title);
  });

});