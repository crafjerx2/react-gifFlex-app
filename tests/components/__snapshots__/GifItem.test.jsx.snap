// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem test suite should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One piece gif"
      data-testid="test-img"
      src="https://media4.giphy.com/media/4OV1bLOIWwIXRxpXlN/giphy.gif?cid=acacc881vaklavil45f31ohfkaui8zwofj1lxi7nsfnuv0y3&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p
      data-testid="test-title"
    >
      One piece gif
    </p>
  </div>
</div>
`;
